sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2Writes.scala

```diff
@@ -45,7 +45,7 @@ object V2Writes extends Rule[LogicalPlan] with PredicateHelper {
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan transformDown {
     case a @ AppendData(r: DataSourceV2Relation, query, options, _, None, _) =>
-      val writeOptions = mergeOptions(options, r.options.asScala.toMap)
+      val writeOptions = mergeOptions(options, r.options.asCaseSensitiveMap.asScala.toMap)
```

Comment:
shall we do the same for other write commands? @huaxingao

Reply:
I will have a follow-up PR for this.